fix issue with changes to struct ip for fbsd 9/10

ViToni opened the pull request:
Submitted by: yuri@rivera.ru
https://svnweb.freebsd.org/ports?view=revision&revision=370646

pali commented:
Hi! Please add proper description into commit message and do not add external links... Also maybe you should squash them into one commit.

ViToni commented:
Could you please check again.

pali commented on src/ifvc.c:
```diff
@@ -268,6 +270,20 @@ void buildIfVc(void) {
+            dp = getIfByName(IfPt->ifr_name, 1);
+            if (dp != NULL && dp->allowednets != NULL) {
+                allowednet = (struct SubnetList *)malloc(sizeof(struct SubnetList));
+                if (allowednet == NULL) my_log(LOG_ERR, 0, "Out of memory !");
```
this is broken code

pali replied:
Please fix this part and also fix your email address in your git commit.

ViToni replied:
Fixed the formatting related to this change.

The email adress is the same as used by changes done inside the GitHub UI. Which one is prefered for consistency (GitHub email or private email)?

pali replied:
Null pointer dereference is still not fixed. Also rebase patch on top of branch.

Email address must be valid. Does not matter if private or public... Just valid address on which people can reach you (e.g. in case somebody would need to update your patch).

ViToni replied:
Maybe you could explain the problem in the above code to me, because my C is quite rusty and I clearly fail to see it.

pali replied:
Null pointer dereference:

allowednet = ...
if (allowednet == NULL) my_log(...);
allowednet->...

ViToni replied:
I believe code above avoids NULL pointer dereference because logging an error via 

> `if (allowednet == NULL) my_log(LOG_ERR, 0, "Out of memory !");`

leads to 'src/syslog.c' and there it exits after logging:

> `if( Severity <= LOG_ERR )
>         exit( -1 );
> `

ViToni commented:
Rewrote all email addresses so that only one email is used for changes by my 
Removed the code in question. The puill request consists now mainly of cleanups, chnage of incompatible code und removal of code duplicates.
Especially the code formatting should ease further patches.

pali commented:
"This branch has conflicts that must be resolved"

And please do different things in different commits/pull requests.
Also include description into commit message. People should understand it without need to open external links...
